app/models/concerns/plans/csv.rb

```diff
@@ -0,0 +1,73 @@
+module Plans::Csv
+  extend ActiveSupport::Concern
+
+  def to_csv business_unit_type: nil
+    options = { col_sep: ';', force_quotes: true, encoding: 'UTF-8' }
+
+    csv_str = CSV.generate(**options) do |csv|
+      csv << csv_headers
+
+      csv_put_business_unit_types_on csv, business_unit_type
+    end
+
+    "\uFEFF#{csv_str}"
+  end
+
+  def csv_filename
+    I18n.t 'plans.csv.csv_name', period: period.name
+  end
+
+  private
+
+    def csv_headers
+      headers = [
+        PlanItem.human_attribute_name(:order_number),
+        PlanItem.human_attribute_name(:status),
+        BusinessUnitType.model_name.human,
+        PlanItem.human_attribute_name(:business_unit_id),
+        PlanItem.human_attribute_name(:project),
+        PlanItem.human_attribute_name(:tags),
+        PlanItem.human_attribute_name(:start),
+        PlanItem.human_attribute_name(:end),
+        (PlanItem.human_attribute_name(:scope) if SHOW_REVIEW_EXTRA_ATTRIBUTES),
+        (PlanItem.human_attribute_name(:risk_exposure) if SHOW_REVIEW_EXTRA_ATTRIBUTES),
+        PlanItem.human_attribute_name(:human_resource_units),
+        PlanItem.human_attribute_name(:material_resource_units),
+        PlanItem.human_attribute_name(:total_resource_units)
+      ].compact
+    end
+
+    def csv_put_business_unit_types_on csv, business_unit_type
+      if business_unit_type
+        csv_rows csv, business_unit_type
+      else
+        business_unit_types.each do |business_unit_type|
+          csv_rows csv, business_unit_type
+        end
+      end
+    end
+
+    def csv_rows csv, business_unit_type
```

Comment:
```suggestion
    def put_csv_rows_on csv, business_unit_type
```
Esto mejor que quede bien explícito en el nombre, sino parece que _devuelve_ columnas, no que las inyecta.